BlackMarket/Managers/ProductsManager.swift

```diff
@@ -0,0 +1,44 @@
+//
+//  ProductsManager.swift
+//  BlackMarket
+//
+//  Created by Santiago Andracchi on 28/09/2023.
+//
+
+import Foundation
+
+final class ProductsManager: ObservableObject {
+  
+  @Published private(set) var products: [Product] = []
+  @Published private(set) var errorMessage: RSAPIError? = nil
```

Comment:
instead of having a single error state at a time(which can lead to race condition and other issues), the public functions can return a Result object or be throwing functions